modules/db/src/test/scala/gem/dao/DaoTest.scala

```diff
@@ -15,6 +15,9 @@ import scala.collection.immutable.TreeMap
 trait DaoTest extends gem.Arbitraries {
   val pid = Program.Id.fromString.unsafeGet("GS-1234A-Q-1")
 
+  private implicit val contextShift: ContextShift[IO] =
+    IO.contextShift(scala.concurrent.ExecutionContext.global)
```

Comment:
What's happening here?

Reply:
`ContextShift[F]` is two things:

1. The ability to reschedule a computation via `shift` … this is kind of like `Thread.sleep()` where you want to relax and let something else have time to make progress, only in this case the work may get picked back up by another thread. `Future` does this every time you map or flatMap; `ContextShift[F]` does it only when you ask. In order to do this it needs an execution context, which it closes over. This is important for scala-js where there's a single event loop and you need to insert `shift` sometimes to ensure fairness. Less of an issue on the JVM.

2. The ability to perform a computation on *another* execution context via `evalOn`, and shift back on completion. doobie uses this to shift all the blocking JDBC primitives onto a dedicated thread pool, then shift back onto the CPU pool when the blocking call completes.

`ContextShift` is not a typeclass because a given program might choose a different execution context, but we assume that it's *unique* per program, which makes it ok to pass implicitly. Think of it as part of the runtime system.

`Timer` is a similar kind of thing, by the way. It gives you a way to "sleep" without blocking and is similarly unique for any given program so we can reasonably pass it implicitly.

`Concurrent` adds on the ability to run things in parallel via `.start` which gives you a fiber back; it's like `forkIO` in Haskell. If you have a `ContextShift[IO]` you get `Concurrent[IO]` for free.

Reply:
Thanks, that makes sense.  I wish _you_ had time and inclination to write the documentation for cats effect!